Apps/AccountDataAccess/src/Patient/PatientMappings.cs

```diff
@@ -0,0 +1,155 @@
+            }
+
+            return null;
+        }
+
+        private static Address? ExtractPostalAddress(PatientIdentity patientIdentity)
+        {
+            if (ShouldReturnPostalAddress(patientIdentity))
+            {
+                List<string> streetLines = new();
+                AddToList(streetLines, patientIdentity.MailAddressStreetOne);
+                AddToList(streetLines, patientIdentity.MailAddressStreetTwo);
+                AddToList(streetLines, patientIdentity.MailAddressStreetThree);
+
+                return new()
+                {
+                    StreetLines = streetLines,
+                    City = patientIdentity.MailAddressCity ?? string.Empty,
+                    PostalCode = patientIdentity.MailAddressPostal ?? string.Empty,
+                    State = patientIdentity.MailAddressProvState ?? string.Empty,
+                    Country = patientIdentity.MailAddressCountry ?? string.Empty,
+                };
+            }
+
+            return null;
+        }
+
+        private static bool ShouldReturnHomeAddress(PatientIdentity patientIdentity)
+        {
+            return !string.IsNullOrWhiteSpace(patientIdentity.HomeAddressStreetOne) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.HomeAddressStreetTwo) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.HomeAddressStreetThree) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.HomeAddressCity) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.HomeAddressPostal) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.HomeAddressProvState) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.HomeAddressCountry);
+        }
+
+        private static bool ShouldReturnPostalAddress(PatientIdentity patientIdentity)
+        {
+            return !string.IsNullOrWhiteSpace(patientIdentity.MailAddressStreetOne) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.MailAddressStreetTwo) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.MailAddressStreetThree) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.MailAddressCity) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.MailAddressPostal) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.MailAddressProvState) ||
+                   !string.IsNullOrWhiteSpace(patientIdentity.MailAddressCountry);
+        }
+
+        private static string? AddToString(string? existingString, string? value)
+        {
+            if (string.IsNullOrWhiteSpace(existingString))
+            {
+                return value;
+            }
+
+            if (string.IsNullOrWhiteSpace(value))
+            {
+                return existingString;
+            }
+
+            return $"{existingString}{Delimiter}{value}";
+        }
+
+        private static void AddToList(List<string> existingList, string? value)
+        {
+            if (value != null)
```

Comment:
```suggestion
            if (!string.IsNullOrEmpty(value))
```
Would this not be a good use of `string.IsNullOrEmpty`? which will still accept whitespaces but not empty strings. If your intention is to accept any non-null value to perfectly replicate the response data, then just ignore this.